src/landscape.cpp

```diff
@@ -634,6 +634,41 @@ void ClearSnowLine()
 	_snow_line = nullptr;
 }
 
+/**
+ * Check if all tiles on the map edge should be considered water borders.
+ * @param allow_non_flat_void Should we allow non-flat void tiles? (if map edge raised, then flattened to sea level)
+ * @return true If the edge of the map is flat and height 0, allowing for infinite water borders.
+ */
+bool CheckWaterBorders(bool allow_non_flat_void)
+{
+	auto check_tile = [&](uint x, uint y, Slope inner_edge) -> bool {
```

Comment:
I don't know a lot about how lambda captures interact with codegen, but since you're only capturing the single boolean, perhaps it's more efficient to explicitly specify that?